docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,121 @@
+# thin-edge Domain Model
+
+The following section introduces the **thin-edge** domain model.
+
+The **thin-edge** domain model explains details of target domains **thin-edge** is primarily designed for.
+It identifies _entities_ and _aspects_ of those target domains, that are touch points for **thin-edge**.
+Additionally it introduces new _entities_ or _aspects_ to seamlessly fit **thin-edge** into those target domains.
+
+Finally the **thin-edge** domain model gives..
+  * _target domain experts_ a clear idea how to position **thin-edge** in their target domain
+  * _target domain experts_ **and** _thin-edge developers_ a common understanding and unique vocabulary for **thin-edge** and it's environment
+
+## thin-edge target domains
+
+**thin-edge** is designed to facilitate IoT functionality to resource constrained **devices**.
+The focus is on industrial OT **devices** or any other kind of embedded **devices**. It is not reduced to **devices** that are capable to install and run thin-edge, but includes also **devices** that need another _(gateway) device_ aside, that executes **thin-edge**.
```

Comment:
I would include IPC, routers or generalized edge devices running linux.

Reply:
> > The focus is on industrial OT **devices**
> 
> I disagree. For me thin-edge focus is on IoT gateways between the cloud and OT devices.

"The focus" does not mean devices that _run_ thin-edge, but those devices that will be _facilitated_ with IoT functionality by thin-edge. So that includes the gateway case, as well as cases where thin-edge can installed directly on an OT device.

So maybe it is more clear when changing wording as below:

    - **thin-edge** is designed to facilitate IoT functionality to resource constrained **devices**. The focus is on industrial OT **devices** or any other kind of embedded **devices**.

    + **thin-edge** is designed to facilitate IoT functionality to resource constrained **devices**, those are primarily industrial OT **devices** or any other kind of embedded **devices**.

Can you go with that, @didier-wenzek  ?

Reply:
> Can you go with that, @didier-wenzek ?

This is not my understanding of the vision for thin-edge.

Reply:
> I disagree. For me thin-edge focus is on IoT gateways between the cloud and OT devices.

Ok, now I think I got your point.

The document explain currently an "OT Device Domain". What you are missing is a cloud perspective, meaning a "Cloud Domain", right?

I will add that.

Reply:
Created now a new section "IoT Cloud Domain". Please have a look.